ciao-cli/client.go

```diff
@@ -315,6 +315,25 @@ func (client *Client) deleteResource(url string, content string) error {
 	return nil
 }
 
+func (client *Client) putResource(url string, content string, request interface{}) error {
+	b, err := json.Marshal(request)
+	if err != nil {
+		return errors.Wrap(err, "Error marshalling JSON")
+	}
+
+	resp, err := client.sendHTTPRequest("PUT", url, nil, bytes.NewReader(b), content)
+	if err != nil {
+		return errors.Wrapf(err, "Error making HTTP request to %s", url)
+	}
+	defer resp.Body.Close()
```

Comment:
No need for the defer.

Reply:
I decided to keep the defer here and removed it from the json unmarhall helper